pkg/assembler/assembler.go

```diff
@@ -17,43 +17,66 @@ package assembler
 
 type assembler struct{}
 
-// Identifiable implements the ability to retrieve a set of
-// attributes such that a graph query is able to identify a
-// GuacNode or GuacEdge uniquely (or as a GuacHyperNode).
-type Identifiable interface {
-	// Identifiers returns a map of fields and values which
-	// can be used to identify an object in the graph.
-	Identifiers() map[string]interface{}
-}
-
 // GuacNode represents a node in the GUAC graph
 type GuacNode interface {
-	Identifiable
-
 	// Type returns the type of node
 	Type() string
 
 	// Properties returns the list of properties of the node
 	Properties() map[string]interface{}
+
+	// Attributes returns the names of the properties of the node.
+	//
+	// If a string `s` is in the list returned by `Attributes` then it
+	// should also be a key in the map returned by `Properties`.
+	Attributes() []string
+
+	// IdentifiableAttributes returns a list of tuples of property names
+	// that can uniquely specify a GuacNode.
+	//
+	// Any string found in a tuple returned by `IdentifiableAttributes`
+	// must also be returned by `Attributes`.
+	IdentifiableAttributes() [][]string
```

Comment:
i think the Identifiable interface does exist here, that would be passed to something that makes the cypher queries. (doesn't have to be a element of this interface, but seems like it could be)

Reply:
So I'm using this one to determine which attributes (in neo4j parlance) / property names (in the way the interface is written) are uniquely identifying a node to avoid duplicated nodes/edges. I'll add a comment describing this.